### PR TITLE
[WIP] Lock down Prometheus Alertmanager and NATS ports by default

### PR DESCRIPTION
**Pull request by johnmccabe**
## Description

This PR makes the following changes:
- Stop exposing the Alertmanager and NATS ports by default on Swarm.
- Add instructions on locking down the Prometheus endpoint to the Nginx contrib docs

## Motivation and Context

- [x] I have raised an issue to propose this change (#356)

## How Has This Been Tested?

**Locking down Alertmanager and NATS**
- Tested auto-scaling still works on Swarm with the Alertmanager port no longer mapped outside of the func network.
- Tested async functions still work on Swarm with the NATS ports no longer mapped outside of the func network.

**TODO**
- [x] test/update behaviour on kubernetes - PR raised openfaas/faas-netes#79
- [ ] update docs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Great idea 👍 if people need these for testing they can re-enable again.

File: docker-compose.extended.yml [hunk 1]
```diff
@@ -50,7 +52,7 @@ services:
                     - 'node.platform.os == linux'
 
     queue-worker:
-        image: functions/queue-worker:0.1.1
+        image: functions/queue-worker:latest-dev
```
Comment (alexellis) [inline docker-compose.extended.yml]: oops.. ^

Comment (johnmccabe) [reply docker-compose.extended.yml]: you ain't seen nothing right, this didn't happen *cough*

Comment (johnmccabe) [reply docker-compose.extended.yml]: Fixed in aba8850

**Comment by alexellis** [comment]
I can't merge due to rebasing error. Happy to take this change though.

**Pull request merged**